src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -235,6 +240,29 @@ protected AuthenticationResult CacheTokenResponseAndCreateAuthenticationResult(M
             }
         }
 
+        internal void CheckResponseFromBroker(MsalTokenResponse msalTokenResponse)
+        {                
+            AuthenticationRequestParameters.RequestContext.Logger.Info(LogMessages.CheckMsalTokenResponseReturnedFromBroker);
+            if(msalTokenResponse.AccessToken != null)
+            {
+                AuthenticationRequestParameters.RequestContext.Logger.Info(
+                    LogMessages.BrokerResponseContainsAccessToken + 
+                    msalTokenResponse.AccessToken.Count());
+                return;
+            }
+            else if(msalTokenResponse.Error != null)
+            {
+                AuthenticationRequestParameters.RequestContext.Logger.Info(
+                    LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
+                throw new MsalServiceException(msalTokenResponse.Error, MsalErrorMessage.BrokerResponseError + msalTokenResponse.ErrorDescription);
+            }
+            else
+            {
+                AuthenticationRequestParameters.RequestContext.Logger.Info(LogMessages.UnknownErrorRerturnedInBrokerResponse);
+                throw new MsalServiceException(MsalError.BrokerReponseReturnedError, MsalErrorMessage.BrokerResponseReturnedError, null);
+            }
+        }
```

Comment:
we have similar logic in ADAL, and we throw an Unknown error here, so i'm just returning a clearer message so we know it's coming from broker. I'm not sure how we would get this response (like the scenario involved) to make the error more specific.